src/Analyzer.Reports/ConsoleReportWriter.cs

```diff
@@ -17,61 +16,59 @@ public class ConsoleReportWriter : IReportWriter
         internal static string IndentedNewLine = Environment.NewLine + "\t";
         internal static string TwiceIndentedNewLine = Environment.NewLine + "\t\t";
 
+        private readonly List<string> filesAlreadyOutput = new List<string>();
+
         /// <inheritdoc/>
         public void WriteResults(IEnumerable<Types.IEvaluation> evaluations, IFileInfo templateFile, IFileInfo parametersFile = null)
         {
-            // Log info on file to be analyzed
-            string fileMetadata = Environment.NewLine + Environment.NewLine + $"File: {templateFile}";
-            if (parametersFile != null)
+            var resultsByFile = ReportsHelper.GetResultsByFile(evaluations, filesAlreadyOutput, out int passedEvaluations);
+
+            // output files in sorted order, but always output root first and regardless if no results
+            var filesWithResults = resultsByFile.Keys.ToList();
+            filesWithResults.Remove(templateFile.FullName);
+            filesWithResults.Sort();
+            filesWithResults.Insert(0, templateFile.FullName);
+
+            foreach(var fileWithResults in filesWithResults)
             {
-                fileMetadata += Environment.NewLine + $"Parameters File: {parametersFile}";
-            }
-            Console.WriteLine(fileMetadata);
+                string fileMetadata;
+                fileMetadata = Environment.NewLine + Environment.NewLine + $"File: {fileWithResults}";
 
-            OutputToConsole(evaluations);
-        }
+                if (fileWithResults == templateFile.FullName)
+                {
+                    if (parametersFile != null)
+                    {
+                        fileMetadata += Environment.NewLine + $"Parameters File: {parametersFile}";
+                    }
+                }
+                else
+                {
+                    fileMetadata += Environment.NewLine + $"Root Template: {templateFile}";
+                }
 
-        private void OutputToConsole(IEnumerable<Types.IEvaluation> evaluations)
```

Comment:
Do we want to keep it as a separate func?

Reply:
The implementation changed and preserving the separate function wasn't as clean or made sense as much to do. Let me know if you have a way to improve readability.